Update the outdated Youtube documentation.

auouymous opened the pull request:
It says 3.10.20 because subscribing to video pages requires gpodder/gpodder#1055. And some other details require latest git. Hopefully, I will get around to a release soon, but need to fix something before translators translate it.

@elelay Is the info about re-installing mac version to update youtube-dl correct?

Fixes gpodder/gpodder#1053.

elelay commented:
Great writeup, thanks!
re-installing the mac app won't update youtube_dl as youtube_dl is added to the .app during gpodder ci, not pip installed from the user's computer. Contrary to windows we can't point to the artifacts produced by the latest ci because circleci hides jobs from the public (tried https://app.circleci.com/pipelines/github/gpodder/gpodder/232/workflows/f62e2766-171c-47c7-8106-7cae5c076045/jobs/612 from a private browsing window redirected to login)

Could you link to [extensions/youtube_dl](https://gpodder.github.io/docs/extensions/youtubedl.html)?

auouymous commented:
Okay, I misunderstood how the new Mac bundle worked, which explains why the option to let users manually update youtube-dl wasn't implemented yet. I changed the page to indicate Mac users have no method of updated youtube-dl. I also added the extension link at first mention, and at the end of the extension section. And I removed mention of the API key from the extension's page.

elelay commented:
Thanks, let's merge it after the release.
